src/jackdaw/client/partitioning.clj

```diff
@@ -79,16 +79,16 @@
 
   See `#'record-key->key-fn` for an example of annotating a topic with
   a `key-fn` function."
-  ([^Producer producer {:keys [topic-name ::key-fn] :as t} value]
+  ([^Producer producer {:keys [key-fn] :as t} value]
```

Comment:
I guess you should change the documentation for this `:key-fn` -> `key-fn` change?

" When constructing a `ProducerRecord` for which no `key` is provided,
  `::key-fn` will be invoked with the record to be produced. "

Reply:
Sure also what this needs is proper testing and use in an example.  This pr is mostly to get things working and create issues to address the missing stuff.